### PR TITLE
move nodepool controller to yurt-manager, and change yurt-manger framework to support mutiversion

### PR DESCRIPTION
**Pull request by kadisi**
#### What type of PR is this?

 /kind feature

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

/assign @rambohe-ch  PTAL

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -36,6 +36,7 @@ require (
 	k8s.io/cluster-bootstrap v0.22.3
 	k8s.io/component-base v0.22.3
 	k8s.io/controller-manager v0.22.3
+	k8s.io/klog v1.0.0
```
Comment (rambohe-ch) [inline go.mod:39]: please use k8s.io/klog/v2 instead of k8s.io/klog

**Comment by rambohe-ch** [comment]
@kadisi please fix golangci-lint error and sonarcloud code analysis error.

File: Makefile [hunk 2]
```diff
@@ -164,10 +164,10 @@ manifests: generate ## Generate WebhookConfiguration, ClusterRole and CustomReso
 # .e.g
 # make newcontroller GROUP=apps VERSION=v1beta1 KIND=example SHORTNAME=examples SCOPE=Namespaced 
 # make newcontroller GROUP=apps VERSION=v1beta1 KIND=example SHORTNAME=examples SCOPE=Cluster
+# make newcontroller GROUP=apps VERSION=v1beta1 KIND=example SHORTNAME=examples SCOPE=Cluster
```
Comment (rambohe-ch) [inline Makefile:167]: 167 line is duplicated with line 166?

File: cmd/yurt-manager/app/manager.go [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+			restclient.SetDefaultWarningHandler(restclient.NoWarnings{})
+			return nil
+		},
+		Run: func(cmd *cobra.Command, args []string) {
+			// verflag.PrintAndExitIfRequested()
+			if s.Generic.Version {
+				fmt.Printf("%s: %#v\n", projectinfo.GetYurtManagerName(), projectinfo.Get())
+				return
+			}
+			fmt.Printf("%s version: %#v\n", projectinfo.GetYurtManagerName(), projectinfo.Get())
```
Comment (rambohe-ch) [inline cmd/yurt-manager/app/manager.go:93]: Line 89~93 can be optimized as following:

```
fmt.Printf("%s version: %#v\n", projectinfo.GetYurtManagerName(), projectinfo.Get())
if s.Generic.Version {
	return
}
```

**Comment by njucjc** [comment]
LGTM

File: cmd/yurt-manager/app/manager.go [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+	}
+
+	fs := cmd.Flags()
+	namedFlagSets := s.Flags()
+	// verflag.AddFlags(namedFlagSets.FlagSet("global"))
+	globalflag.AddGlobalFlags(namedFlagSets.FlagSet("global"), cmd.Name())
+	// hoist this flag from the global flagset to preserve the commandline until
+	// the gce cloudprovider is removed.
+	// globalflag.Register(namedFlagSets.FlagSet("generic"), "cloud-provider-gce-lb-src-cidrs")
+	// namedFlagSets.FlagSet("generic").MarkDeprecated("cloud-provider-gce-lb-src-cidrs", "This flag will be removed once the GCE Cloud Provider is removed from kube-controller-manager")
```
Comment (rambohe-ch) [inline cmd/yurt-manager/app/manager.go:125]: please cleanup the commented lines 120, 122~125

File: pkg/apis/apps/v1beta1/default.go [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+)
+
+// SetDefaultsNodePool set default values for NodePool.
+func SetDefaultsNodePool(obj *NodePool) {
+	// example for set default value for NodePool
+	if obj.Annotations == nil {
+		obj.Annotations = make(map[string]string)
+	}
+
+	obj.Annotations["test"] = "test"
```
Comment (rambohe-ch) [inline pkg/apis/apps/v1beta1/default.go:33]: please remove testing code.

File: pkg/controller/nodepool/nodepool_controller.go [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+// Reconcile reads that state of the cluster for a NodePool object and makes changes based on the state read
+// and what is in the NodePool.Spec
+func (r *ReconcileNodePool) Reconcile(_ context.Context, req reconcile.Request) (reconcile.Result, error) {
+
+	// Note !!!!!!!!!!
+	// We strongly recommend use Format() to  encapsulation because Format() can print logs by module
+	// @kadisi
+	klog.Infof(Format("Reconcile NodePool %s/%s", req.Namespace, req.Name))
+
+	ctx := context.Background()
```
Comment (DrmagicE) [inline pkg/controller/nodepool/nodepool_controller.go:147]: How about using the context in the `Reconcile` input?

Comment (kadisi) [reply pkg/controller/nodepool/nodepool_controller.go:147]: good, thanks

**Review by rambohe-ch — APPROVED** [review]

**Pull request merged**